addons/event_sale/models/sale_order.py

```diff
@@ -47,6 +47,13 @@ def _compute_attendee_count(self):
         for sale_order in self:
             sale_order.attendee_count = attendee_count_data.get(sale_order.id, 0)
 
+    def action_cancel(self):
+        super(SaleOrder, self).action_cancel()
+        for sale_order in self:
+            event_registrations = self.env['event.registration'].search([('sale_order_id', '=', sale_order.id)])
+            if event_registrations:
+                event_registrations.action_cancel()
```

Comment:
But it appears that the `toggle_active` method does not get called when clicking on the cancel button of the SO.

In this case how can we use `toggle_active` here?

Reply:
My bad, I thought "action_cancel" was archiving the record but not at all, forget what I said.

This could be improved a bit then:

```py
def action_cancel(self):
    super(SaleOrder, self).action_cancel()
    if self.ids:
        self.env['event.registration'].search([('sale_order_id', 'in', self.ids)]).action_cancel()
```

Reply:
Okay, It has now been improved with your suggestion.